pkgs/applications/office/pympress/default.nix

```diff
@@ -0,0 +1,48 @@
+{ lib, python3Packages, pkgs, vlc }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "pympress";
+  version = "1.2.0";
+
+  src = python3Packages.fetchPypi {
+    inherit pname version;
+    sha256 = "10v1857kbd57s3iq5b9hb5k2ivhx5jcmjzjc0d3lw9mvl280yg86";
+  };
+
+  /* with check, segmentation fault:
+  _gtk_style_provider_private_get_settings: assertion 'GTK_IS_STYLE_PROVIDER_PRIVATE (provider)' failed
+  */
+  doCheck = false;
+
+  nativeBuildInputs = with pkgs; [
+    wrapGAppsHook
+  ];
+
+  buildInputs = with pkgs; [
+    gtk3
+    gobject-introspection
+    libcanberra-gtk3
+    (poppler.override {introspectionSupport = true;})
+    vlc
+  ];
+
+  propagatedBuildInputs =  with python3Packages; [
+    pycairo
+    pygobject3
+    python-vlc
+  ];
+
+  postInstall = ''
+    wrapProgram $out/bin/pympress \
```

Comment:
It might be better to patch python-vlc a source:

https://github.com/oaubert/python-vlc/blob/b61a1c1840f470ed36f36c41040ef3479acd29b8/generated/dev/vlc.py#L193

We do that for other bindings:

https://github.com/NixOS/nixpkgs/blob/7a91369508f1d03c343e4a31f73149d84892c6bf/pkgs/development/python-modules/pyogg/default.nix#L17-L28

Reply:
hmm, I'm not sure that `wrapProgram` is the right approach but I'm skeptical about changing the source, that seems hard to maintain. https://docs.python.org/3/library/ctypes.html#finding-shared-libraries shows that `LD_LIBRARY_PATH` is used, so just need to get vlc on that I believe...Is there another way to set `LD_LIBRARY_PATH`? It seems this is why the `nix-shell` approach works in https://github.com/NixOS/nixpkgs/pull/62137#issuecomment-496641317 ?

Reply:
Yeah, `cDLL` works similarly to `dlopen`. The issue with environment variables is that they contaminate subprocesses, and that they need to be repeated in every consumer which is tiresome and breaks encapsulation.

Maintenance cost is quite minor in practice. The source code loading the library does not usually change very often and when it does it is good to be notified about the failure since it often indicates a switch to a different library version.

I do not think setting `LD_LIBRARY_PATH` behaves any differently in `nix-shell`. You would be running wrapped binary either way.

Reply:
are you able to reproduce https://github.com/NixOS/nixpkgs/pull/62137#issuecomment-496641317 using the files below? I can do the source patching as you suggest, but I'd love to understand why `libvlc.so.5` can be found inside a nix-shell but not when the executable is run directly as perhaps if this is understood we can avoid source paching